src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -263,30 +263,49 @@ IEnumerable<ExpressionSyntax> assignments
 			var allAssignmentsAreOfImmutableValues = true;
 
 			foreach( var assignment in assignments ) {
-				var query = GetImmutabilityQueryForAssignment( assignment );
-
-				if( query == null ) {
-					// null is a signal that there is nothing further that needs to
-					// be checked for this assignment.
-					continue;
-				}
-
-				if( m_context.IsImmutable(
-					query.Value,
-					() => assignment.GetLocation(),
-					out var diagnostic
-				) ) {
-					continue;
+				var kind = GetQueryForAssignment( assignment, out var query, out var diagnostic );
+
+				switch( kind ) {
+					case AssignmentQueryKind.NothingToCheck:
+						continue;
+
+					case AssignmentQueryKind.Hopeless:
+						diagnosticSink( diagnostic );
+						allAssignmentsAreOfImmutableValues = false;
+						continue;
+
+					case AssignmentQueryKind.ImmutabilityQuery:
+						if( m_context.IsImmutable(
+							query,
+							() => assignment.Expression.GetLocation(),
+							out diagnostic
+						) ) {
+							continue;
+						}
+
+						diagnosticSink( diagnostic );
+
+						// We're going to keep looking at all writes to surface as many
+						// relevant diagnostics as possible.
+						allAssignmentsAreOfImmutableValues = false;
+						continue;
 				}
+			}
 
-				diagnosticSink( diagnostic );
+			return isReadOnly && allAssignmentsAreOfImmutableValues;
+		}
 
-				// We're going to keep looking at all writes to surface as many
-				// relevant diagnostics as possible.
-				allAssignmentsAreOfImmutableValues = false;
+		private readonly struct AssignmentInfo {

```

Comment:
RE: https://github.com/Brightspace/D2L.CodeStyle/pull/729#discussion_r591675842
Now just keeping track of which one is the initializer.
This is a little lame since there is at most one but 🤷🏼‍♂️